massa-grpc/src/public.rs

```diff
@@ -0,0 +1,736 @@
+// Copyright (c) 2023 MASSA LABS <info@massa.net>
+
+use crate::error::GrpcError;
+use crate::server::MassaPublicGrpc;
+
+use massa_execution_exports::mapping_grpc::{
+    to_event_filter, to_execution_query_response, to_querystate_filter,
+};
+use massa_execution_exports::{
+    ExecutionQueryRequest, ExecutionStackElement, ReadOnlyExecutionRequest, ReadOnlyExecutionTarget,
+};
+use massa_models::address::Address;
+use massa_models::block::Block;
+use massa_models::block_id::BlockId;
+use massa_models::config::CompactConfig;
+use massa_models::operation::{OperationId, SecureShareOperation};
+use massa_models::prehash::PreHashSet;
+use massa_models::slot::Slot;
+use massa_models::timeslots::get_latest_block_slot_at_timestamp;
+use massa_proto_rs::massa::api::v1 as grpc_api;
+use massa_proto_rs::massa::model::v1::{self as grpc_model, read_only_execution_call};
+use massa_time::MassaTime;
+use std::collections::{BTreeMap, HashSet};
+use std::str::FromStr;
+use tracing::log::warn;
+
+/// Execute read only call (function or bytecode)
+pub(crate) fn execute_read_only_call(
+    grpc: &MassaPublicGrpc,
+    request: tonic::Request<grpc_api::ExecuteReadOnlyCallRequest>,
+) -> Result<grpc_api::ExecuteReadOnlyCallResponse, GrpcError> {
+    let inner_req = request.into_inner();
+    let call: grpc_model::ReadOnlyExecutionCall = inner_req
+        .call
+        .ok_or_else(|| GrpcError::InvalidArgument("no call provided".to_string()))?;
+
+    let mut call_stack = Vec::new();
+
+    let target = if let Some(call_target) = call.target {
+        match call_target {
+            read_only_execution_call::Target::BytecodeCall(value) => {
+                let caller_address = Address::from_str(&call.caller_address)?;
+                // TODO check if we keep the old format of operation datastore
```

Comment:
@AurelienFT should we keep Massa derializer ? or just keep protobuf way ?

Reply:
I don't understand the question 😆

Reply:
The Massa deserialiser way 

https://github.com/massalabs/massa/blob/7c5e860deec2fcad14de01f55ffb31ef7e4df276/massa-api/src/public.rs#L152-L171

Reply:
The protobuf way 

https://github.com/massalabs/massa/blob/7c5e860deec2fcad14de01f55ffb31ef7e4df276/massa-grpc/src/public.rs#L47-L51

Reply:
The first is safer

Reply:
done with the safest way